python/domain_decomposition.cpp

```diff
@@ -0,0 +1,92 @@
+#include <string>
+#include <sstream>
+
+#include <pybind11/pybind11.h>
+
+#include <arbor/context.hpp>
+#include <arbor/domain_decomposition.hpp>
+#include <arbor/load_balance.hpp>
+
+#include "context.hpp"
+#include "recipe.hpp"
+#include "strprintf.hpp"
+
+namespace pyarb {
+
+std::string gd_string(const arb::group_description& g) {
+    std::stringstream s;
+    s << "<cell group: " << g.gids.size()
+      << " cells, gids [" << util::csv(g.gids, 5) << "]"
+      << ", " << g.kind << ", " << g.backend << ">";
+    return s.str();
+}
+
+std::string dd_string(const arb::domain_decomposition& d) {
+    std::stringstream s;
+    s << "<domain decomposition: domain "
+      << d.domain_id << " of "
+      << d.num_domains << ", "
+      << d.num_local_cells << "/" << d.num_global_cells << " loc/glob cells, "
+      << d.groups.size() << " groups>";
+
+    return s.str();
+}
+
+void register_domain_decomposition(pybind11::module& m) {
+    using namespace pybind11::literals;
+
+    // Group description
+    pybind11::class_<arb::group_description> group_description(m, "group_description",
+        "The indexes of a set of cells of the same kind that are grouped together in a cell group.");
+    group_description
+        .def(pybind11::init<arb::cell_kind, std::vector<arb::cell_gid_type>, arb::backend_kind>(),
```

Comment:
I see what you were talking about yesterday: if we provide an interface for users to construct a `group_description`, that interface must enforce preconditions, e.g. "all gid must be sorted".
I will check whether that precondition still applies: I suspect that we actually removed that precondition at some point (I can remember changing all the cell group implementations to use a hash table for looking up gid information, which doesn't require sorted input).